upup/pkg/fi/cloudup/awsup/aws_cloud.go

```diff
@@ -749,3 +753,113 @@ func (c *awsCloudImplementation) FindVPCInfo(vpcID string) (*fi.VPCInfo, error)
+	//MasterMachineType: n1-standard-16
+	//{{ else if gt .TotalNodeCount 100 }}
+	//MasterMachineType: n1-standard-8
+	//{{ else if gt .TotalNodeCount 10 }}
+	//MasterMachineType: n1-standard-4
+	//{{ else if gt .TotalNodeCount 5 }}
+	//MasterMachineType: n1-standard-2
+	//{{ else }}
+	//MasterMachineType: n1-standard-1
+	//{{ end }}
+	//
+	//{{ if gt TotalNodeCount 500 }}
+	//MasterMachineType: c4.8xlarge
+	//{{ else if gt TotalNodeCount 250 }}
+	//MasterMachineType: c4.4xlarge
+	//{{ else if gt TotalNodeCount 100 }}
+	//MasterMachineType: m3.2xlarge
+	//{{ else if gt TotalNodeCount 10 }}
+	//MasterMachineType: m3.xlarge
+	//{{ else if gt TotalNodeCount 5 }}
+	//MasterMachineType: m3.large
+	//{{ else }}
+	//MasterMachineType: m3.medium
+	//{{ end }}
+
+	case kops.InstanceGroupRoleNode:
+		candidates = []string{"t2.medium"}
+
+	case kops.InstanceGroupRoleBastion:
+		candidates = []string{"t2.micro"}
+
+	default:
+		return "", fmt.Errorf("unhandled role %q", ig.Spec.Role)
+	}
+
+	// Find the AZs the InstanceGroup targets
+	igZones := sets.NewString()
+	for _, subnetName := range ig.Spec.Subnets {
+		var subnet *kops.ClusterSubnetSpec
+		for i := range cluster.Spec.Subnets {
+			if cluster.Spec.Subnets[i].Name == subnetName {
+				subnet = &cluster.Spec.Subnets[i]
+			}
+		}
+		if subnet == nil {
+			return "", fmt.Errorf("subnet %q is not defined in cluster", subnetName)
+		}
+		igZones.Insert(subnet.Zone)
+	}
+
+	// TODO: Validate that instance type exists in all AZs, but skip AZs that don't support any VPC stuff
+	for _, instanceType := range candidates {
+		zones, err := c.zonesWithInstanceType(instanceType)
+		if err != nil {
+			return "", err
+		}
+		if zones.IsSuperset(igZones) {
+			return instanceType, nil
+		} else {
+			glog.V(2).Infof("can't use instance type %q, available in zones %v but need %v", instanceType, zones, igZones)
+		}
+	}
+
+	return "", fmt.Errorf("could not find a suitable supported instance type for the instance group %q (type %q) in region %q", ig.Name, ig.Spec.Role, c.region)
+}
+
+// supportsInstanceType uses the DescribeReservedInstancesOfferings API call to determine if an instance type is supported in a region
+func (c *awsCloudImplementation) zonesWithInstanceType(instanceType string) (sets.String, error) {
+	glog.V(4).Infof("checking if instance type %q is supported in region %q", instanceType, c.region)
+	request := &ec2.DescribeReservedInstancesOfferingsInput{}
+	request.InstanceTenancy = aws.String("default")
+	request.IncludeMarketplace = aws.Bool(false)
+	request.OfferingClass = aws.String("standard")
+	request.OfferingType = aws.String("No Upfront")
+	request.ProductDescription = aws.String("Linux/UNIX (Amazon VPC)")
+	request.InstanceType = aws.String(instanceType)
+
+	zones := sets.NewString()
+
+	response, err := c.ec2.DescribeReservedInstancesOfferings(request)
+	if err != nil {
+		return zones, fmt.Errorf("error checking if instance type %q is supported in region %q: %v", instanceType, c.region, err)
```

Comment:
return a list of what is available?  Not a big deal, and not for this PR.  Just a thought